onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -1375,6 +1376,37 @@ ORT_API_STATUS_IMPL(OrtApis::GetOpaqueValue, _In_ const char* domain_name, _In_
   return nullptr;
 }
 
+ORT_API_STATUS_IMPL(OrtApis::GetAvailableProviders, _Outptr_ char ***out_ptr,
+                    _In_ int *providers_length) {
+  API_IMPL_BEGIN
+  const int MAX_LEN = 30;
+  int available_count = (int)(sizeof(providers_available) / sizeof(char *));
+  char **out = (char **)malloc(available_count * sizeof(char *));
+  for(int i = 0; i < available_count; i++) {
+      out[i] = (char *)malloc(
+        strnlen(providers_available[i], MAX_LEN) * sizeof(char));
+#ifdef _MSC_VER
+      strncpy_s(out[i], MAX_LEN, providers_available[i], MAX_LEN);
+#else
+      strncpy(out[i], providers_available[i], MAX_LEN);
```

Comment:
That's wrong. No null-character is implicitly appended at the end of destination if source is longer than num. 
Please try strdup or manually set '\0' there.

Reply:
You are right. I don't think strdup() is in the C standard library. I also need to add check whether malloc() succeded.

Reply:
Fixed.